codegenerator/cli/templates/static/codegen/src/db/EntityHistory.res

```diff
@@ -0,0 +1,325 @@
+open Table
+
+type historyFieldsGeneral<'a> = {
+  chain_id: 'a,
+  block_timestamp: 'a,
+  block_number: 'a,
+  log_index: 'a,
+}
+
+type historyFields = historyFieldsGeneral<int>
+
+type entityIdOnly = {id: string}
+let entityIdOnlySchema = S.schema(s => {id: s.matches(S.string)})
+type entityData<'entity> = Delete(entityIdOnly) | Set('entity)
```

Comment:
Entity data can now be delete or set. Delete still needs the ID, and the only way we can distingish between an entity that already has all nullable fields and a deleted entity is with an additional flag (which is added to the table.)